docker/puppet-agent-alpine/Dockerfile

```diff
@@ -0,0 +1,130 @@
+FROM alpine:3.8 as build
+
+RUN apk add --no-cache cmake boost-dev make curl git curl-dev ruby ruby-dev gcc g++ yaml-cpp-dev
+
+RUN mkdir /workspace 
+WORKDIR /workspace
+RUN git clone -b 1.5.x https://github.com/puppetlabs/leatherman && \
```

Comment:
Why do this in one layer since it's a build image? You're throwing away each layer if any component changes.

Reply:
Good point, will rework